ignite/pkg/cosmosbuf/buf.go

```diff
@@ -67,6 +73,55 @@ func (c Command) String() string {
 	return string(c)
 }
 
+// Export runs the buf Export command for the files in the proto directory.
+func (b Buf) Export(ctx context.Context, protoDir, output string) error {
+	// Check if the proto directory is the Cosmos SDK one
+	if strings.Contains(protoDir, cosmosver.CosmosModulePath) {
+		if b.sdkProtoDir == "" {
+			// Copy Cosmos SDK proto path without the Buf workspace.
+			// This is done because the workspace contains a reference to
+			// a "orm/internal" proto folder that is not present by default
+			// in the SDK repository.
+			d, err := CopySDKProtoDir(protoDir)
+			if err != nil {
+				return err
+			}
+
+			b.sdkProtoDir = d
+		}
+
+		// Split absolute path into an absolute prefix and a relative suffix
+		paths := strings.Split(protoDir, "/proto")
+		if len(paths) < 2 {
+			return fmt.Errorf("invalid Cosmos SDK mod path: %s", protoDir)
+		}
+
+		// Use the SDK copy to resolve SDK proto files
+		protoDir = filepath.Join(b.sdkProtoDir, paths[1])
+	}
+
+	if err := checkContainsProtoFiles(protoDir); err != nil {
+		return err
+	}
+
+	flags := map[string]string{
+		flagOutput: output,
+	}
+
+	cmd, err := b.generateCommand(CMDExport, flags, protoDir)
+	if err != nil {
+		return err
+	}
+
+	g, ctx := errgroup.WithContext(ctx)
```

Comment:
Why use an errorgroup and a goroutine if the main threat is blocking waiting for the goroutine

Reply:
fixed